### PR TITLE
Initial Discord Rich Presence support

### PR DESCRIPTION
**Pull request by occune**
but we still need to tweak cosmetic aspects (like gamestates etc) and change the timer
huge thanks to @Injazz

### CODE REVIEW

File: BuildChecker/BuildChecker.csproj [hunk 1]
```diff
@@ -33,6 +33,19 @@ https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild
   <PropertyGroup>
     <StartupObject />
   </PropertyGroup>
+  <ItemGroup>
```
Comment (PJB3005) [inline BuildChecker/BuildChecker.csproj:36]: I think you added these references to the RPC assembly in a bit too much projects there. It should only be in `SS14.Client`

File: SS14.Client/GameController.cs [hunk 2]
```diff
@@ -155,6 +157,8 @@ private void Startup()
             _placementManager.Initialize();
             _viewVariablesManager.Initialize();
 
+            RichPresenceClient presence = new RichPresenceClient();
```
Comment (PJB3005) [inline SS14.Client/GameController.cs:160]: This should be made an IoC dependency:

https://github.com/space-wizards/space-station-14/wiki/IoC-(Inversion-of-Control)

File: SS14.Client/Utility/RichPresenceClient.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            timer.Start();
+
+            //Connect
+            client.Initialize();
+
+            //Send a presence. Do this as many times as you want
+            client.SetPresence(new RichPresence()
+            {
+                Details = "Gamemode: Valid Hunt Racing",
+                State = "/vg/station",
```
Comment (PJB3005) [inline SS14.Client/Utility/RichPresenceClient.cs:36]: Not sure whether referring to /vg/ directly is a good idea.

I think a more agnostic name might be better.

File: SS14.Client/Utility/RichPresenceClient.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                Assets = new Assets()
+                {
+                    LargeImageKey = "vgstation-logo2",
+                    LargeImageText = "Meta Station",
+                    SmallImageKey = "logo",
+                },
+                Timestamps = Timestamps.FromTimeSpan(10)
```
Comment (PJB3005) [inline SS14.Client/Utility/RichPresenceClient.cs:43]: Shouldn't this value actually be incremented periodically? If it isn't wouldn't it be better to leave it out?

File: SS14.Client/Utility/RichPresenceClient.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            });       
+        }
+
+        public void Free()
```
Comment (PJB3005) [inline SS14.Client/Utility/RichPresenceClient.cs:47]: You should implement `IDisposable` and rename this to `Dispose()`.

**Review by PJB3005 — CHANGES_REQUESTED** [review]
I guess it would be nice to have some things like changing the status when connecting to a server but oh well.

**Comment by io-june** [comment]
I also want to note that there is a bug in DiscordRPC causing discord pipe to fail connection on Mono.
https://github.com/Lachee/discord-rpc-csharp/issues/54
you need fresh nuget version (which gonna update soon), or do it manually

File: SS14.Client/GameController.cs (hunk 2 above)
Comment (ZoldorfTheWizard) [reply SS14.Client/GameController.cs:160]: +1 ^

**Comment by PJB3005** [comment]
Superseded by #776

**Pull request closed without merging**